gke/autopilot/reservation/main.tf

```diff
@@ -0,0 +1,195 @@
+/**
+* Copyright 2024 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+data "google_client_config" "default" {}
+
+resource "google_container_cluster" "default" {
+  name     = "gke-autopilot-cluster"
+  location = "us-central1"
+
+  enable_autopilot = true
+
+  # Set `deletion_protection` to `true` will ensure that one cannot
+  # accidentally delete this instance by use of Terraform.
+  deletion_protection = false
+}
+
+# [START gke_autopilot_reservation_specific_reservation]
+resource "google_compute_reservation" "specific_pod" {
+  name = "specific-reservation-pod"
+  zone = "us-central1-a"
+
+  specific_reservation {
+    count = 1
+
+    instance_properties {
+      machine_type = "c3-standard-4-lssd"
+
+      local_ssds {
+        disk_size_gb = 375
+        interface    = "NVME"
+      }
+    }
+  }
+
+  specific_reservation_required = true
+}
+# [END gke_autopilot_reservation_specific_reservation]
+
+# [START gke_autopilot_reservation_specific_cluster]
+resource "google_compute_reservation" "specific_accelerator" {
+  name = "specific-reservation-accelerator"
+  zone = "us-central1-a"
```

Comment:
the cluster is regional, but the reservation is for a zone, not sure if it works.

Reply:
Good Catch @yliaog! - Worked fine during testing, but I'll adjust for consistency and re-test.

Reply:
OK, so `google_compute_reservation` is always zone specific, and Autopilot clusters are regional, so this appears to be working correctly/intended.